analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -43,9 +44,77 @@ protected override ProgramState[] Process(SymbolicContext context, IInvocationOp
         {
             state = state.ResetFieldConstraints();
         }
-        return invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace))
-            ? ProcessStringIsNullOrEmpty(context, invocation)
-            : new[] { state };
+        if (invocation.TargetMethod.ContainingType.Is(KnownType.System_Linq_Enumerable))
+        {
+            return ProcessLinqEnumerable(context, invocation);
+        }
+        else if (invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace)))
+        {
+            return ProcessStringIsNullOrEmpty(context, invocation);
+        }
+        else
+        {
+            return new[] { state };
+        }
+    }
+
+    private static ProgramState[] ProcessLinqEnumerable(SymbolicContext context, IInvocationOperationWrapper invocation)
+    {
+        switch (invocation.TargetMethod.Name)
+        {
+            case "Append":
+            case nameof(Enumerable.AsEnumerable):
+            case nameof(Enumerable.Cast):
+            case "Chunk":
+            case nameof(Enumerable.Concat):
+            case nameof(Enumerable.Distinct):
+            case "DistinctBy":
+            case nameof(Enumerable.Empty):
+            case nameof(Enumerable.Except):
```

Comment:
ExceptBy

Reply:
In preview

Reply:
It's in .Net 6
![image](https://user-images.githubusercontent.com/103252490/192463105-2505c4d0-de6c-41f7-a8e1-299f8546cbb8.png)
Can't we test this?

Reply:
Indeed